app/models/log.rb

```diff
@@ -4,11 +4,25 @@ class Log < ApplicationRecord
   belongs_to :owning_organisation, class_name: "Organisation", optional: true
   belongs_to :managing_organisation, class_name: "Organisation", optional: true
   belongs_to :created_by, class_name: "User", optional: true
+  before_save :update_status!
 
   STATUS = { "not_started" => 0, "in_progress" => 1, "completed" => 2 }.freeze
   enum status: STATUS
 
-  scope :combined, -> { ActiveRecord::Base.connection.execute("SELECT * FROM logs") }
+  scope :filter_by_organisation, ->(org, _user = nil) { where(owning_organisation: org).or(where(managing_organisation: org)) }
+  scope :filter_by_status, ->(status, _user = nil) { where status: }
+  scope :filter_by_years, lambda { |years, _user = nil|
+    first_year = years.shift
+    query = filter_by_year(first_year)
+    years.each { |year| query = query.or(filter_by_year(year)) }
+    query.all
+  }
+  scope :filter_by_id, ->(id) { where(id:) }
+  scope :filter_by_user, lambda { |selected_user, user|
+    if !selected_user.include?("all") && user.present?
```

Comment:
What is `all` and why do we need to filter it out?

Reply:
It's a filter. We can select either "All" or "Yours" when we're filtering the logs by user. This is not a new implementation, just moved it to parent class